src/overlays/actors/ovl_En_Mkk/z_en_mkk.c

```diff
@@ -82,62 +111,498 @@ static DamageTable D_80A4F754 = {
+
+    this->unk_14C = CLAMP(this->actor.shape.rot.z, 0, 16);
+
+    this->actor.shape.rot.z = 0;
+    this->actor.world.rot.z = 0;
+    this->actor.shape.rot.x = 0;
+    this->actor.world.rot.x = 0;
+
+    Math_Vec3f_Copy(&this->unk_154, &this->actor.world.pos);
+    Math_Vec3f_Copy(&this->unk_160, &this->actor.world.pos);
+    this->unk_14B = (this->actor.params & 4) ? 8 : 0;
+
+    params8 = (this->actor.params >> 8) & 0xFF;
+    params2 = this->actor.params & 2;
+    this->actor.params &= 1;
+    if (this->actor.params == 1) {
+        this->actor.hintId = 0x3C;
+    } else {
+        this->actor.hintId = 0x2C;
+    }
+    if ((params8 == 0) || (params8 == 255)) {
+        func_80A4E0CC(this);
+        this->unk_178 = 30000.0f;
+        if (params2 > 0) {
+            this->unk_14B |= 4;
+        }
+    } else {
+        this->unk_178 = params8 * 40.0f * 0.1f;
+        func_80A4EDF0(this);
+        this->unk_14E = 0;
+    }
+}
+
+void EnMkk_Destroy(Actor* thisx, PlayState* play) {
+    EnMkk* this = THIS;
+
+    Collider_DestroySphere(play, &this->collider);
+}
+
+void func_80A4E0CC(EnMkk* this) {
+    this->alpha = 0;
+    this->unk_14B |= 3;
+    this->actor.flags &= ~ACTOR_FLAG_1;
+    this->actionFunc = func_80A4E100;
+}
+
+void func_80A4E100(EnMkk* this, PlayState* play) {
+    s32 newAlpha;
+
+    if ((this->unk_14B & 4) != 0) {
+        newAlpha = this->alpha + 15;
+    } else {
+        newAlpha = this->alpha + 5;
+    }
+    if (newAlpha >= 255) {
+        this->primColorSelect = 3;
+        this->collider.base.acFlags |= AC_ON;
+        this->alpha = 255;
+        this->actor.flags |= ACTOR_FLAG_1;
+        this->actor.shape.rot.y = this->actor.yawTowardsPlayer;
+        this->unk_14B &= ~4;
+        func_80A4E190(this);
+    } else {
+        this->alpha = newAlpha;
+    }
+}
+
+void func_80A4E190(EnMkk* this) {
+    this->unk_14E = (s32)(Rand_ZeroOne() * 20.0f) + 10;
+    if (this->unk_149 != 0) {
+        this->unk_149--;
+    }
+    this->actionFunc = func_80A4E1F0;
+}
+
+void func_80A4E1F0(EnMkk* this, PlayState* play) {
+    this->unk_14E--;
+    if (this->unk_14E == 0) {
+        func_80A4E2B8(this);
+    }
+}
+
+void func_80A4E22C(EnMkk* this, PlayState* play) {
+    Vec3f pos;
+
+    pos.x = this->actor.world.pos.x;
+    pos.y = this->actor.world.pos.y + 15.0f;
+    pos.z = this->actor.world.pos.z;
+    func_800B0DE0(play, &pos, &gZeroVec3f, &D_80A4F788, &D_80A4F780, &D_80A4F784, 350, 20);
+}
+
+void func_80A4E2B8(EnMkk* this) {
+    this->unk_14E = 30;
+    this->unk_150 = this->actor.shape.rot.y;
+    this->unk_14B &= ~1;
+    this->actionFunc = func_80A4E2E8;
+}
+
+void func_80A4E2E8(EnMkk* this, PlayState* play) {
+    Player* sp24 = GET_PLAYER(play);
+    s32 sp20;
+
+    this->unk_14E--;
+    if ((this->actor.params == 1) && (this->actor.bgCheckFlags & 1) && (this->actor.speedXZ > 2.5f) &&
+        (!(play->gameplayFrames % 3))) {
```

Comment:
```suggestion
        ((play->gameplayFrames % 3) == 0)) {
```
`!` only really applies to bools and flags